src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -500,137 +545,181 @@ private BoundPattern BindRecursivePattern(RecursivePatternSyntax node, TypeSymbo
-            var deconstruction = node.DeconstructionPatternClause;
-            var patterns = ArrayBuilder<BoundSubpattern>.GetInstance(deconstruction.Subpatterns.Count);
-            foreach (var subpatternSyntax in deconstruction.Subpatterns)
+            foreach (var subpatternSyntax in node.Subpatterns)
             {
-                TypeSymbol elementType = objectType;
                 if (subpatternSyntax.NameColon != null)
                 {
                     // error: name not permitted in ITuple deconstruction
                     diagnostics.Add(ErrorCode.ERR_ArgumentNameInITuplePattern, subpatternSyntax.NameColon.Location);
                 }
 
-                // Since the input has been cast to ITuple, it must be escapable.
-                const uint valEscape = Binder.ExternalScope;
                 var boundSubpattern = new BoundSubpattern(
                     subpatternSyntax,
                     null,
-                    BindPattern(subpatternSyntax.Pattern, elementType, valEscape, false, diagnostics));
+                    BindPattern(subpatternSyntax.Pattern, objectType, valEscape, false, diagnostics));
                 patterns.Add(boundSubpattern);
             }
+        }
 
-            return new BoundITuplePattern(node, iTupleGetLength, iTupleGetItem, patterns.ToImmutableAndFree(), inputType, hasErrors);
+        private void BindITupleSubpatterns(
+            ParenthesizedVariableDesignationSyntax node,
+            ArrayBuilder<BoundSubpattern> patterns,
+            DiagnosticBag diagnostics)
+        {
+            // Since the input has been cast to ITuple, it must be escapable.
+            const uint valEscape = Binder.ExternalScope;
+            var objectType = Compilation.GetSpecialType(SpecialType.System_Object);
+            foreach (var variable in node.Variables)
+            {
+                BoundPattern pattern = BindVarDesignation(variable, objectType, valEscape, false, diagnostics);
+                var boundSubpattern = new BoundSubpattern(
+                    variable,
+                    null,
+                    pattern);
+                patterns.Add(boundSubpattern);
+            }
         }
 
-        private ImmutableArray<BoundSubpattern> BindDeconstructionPatternClause(
+        private void BindValueTupleSubpatterns(
             DeconstructionPatternClauseSyntax node,
             TypeSymbol declType,
             uint inputValEscape,
-            DiagnosticBag diagnostics,
-            out MethodSymbol deconstructMethod,
-            ref bool hasErrors)
+            ref bool hasErrors,
+            ArrayBuilder<BoundSubpattern> patterns,
+            DiagnosticBag diagnostics)
         {
-            deconstructMethod = null;
-            var patterns = ArrayBuilder<BoundSubpattern>.GetInstance(node.Subpatterns.Count);
-            if (declType.IsTupleType)
+            ImmutableArray<TypeSymbolWithAnnotations> elementTypes = declType.TupleElementTypes;
```

Comment:
>elementTypes [](start = 54, length = 12)

If `elementTypes` is passed in as a parameter, could `BoundValueTupleSubpatterns` be used for 0-element tuples as well?